src/Playgrounds/CosmosEndToEnd/CosmosEndToEnd.ApiService/Program.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+using CosmosEndToEnd.ApiService;
+using Microsoft.Azure.Cosmos;
+
+var sessionId = Guid.NewGuid().ToString();
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.AddServiceDefaults();
+builder.AddAzureCosmosDB("db",
+                         settings =>
+                            {
+                                settings.IgnoreEmulatorCertificate = true;
+                            },
+                         clientOptions =>
+                         {
+                             // Default serializer for Cosmos V3 client is JSON.NET, this changes
+                             // us to use S.T.J for this playground.
+                             clientOptions.Serializer = new StjSerializer(new System.Text.Json.JsonSerializerOptions());
+                         });
+
+builder.Services.AddKeyedSingleton<Container>("entries", (sp, _) =>
+{
+    var db = sp.GetRequiredService<Database>();
+    var container = db.CreateContainerIfNotExistsAsync("entries", "/sessionId").Result.Container;
+    return container;
+});
+
+var app = builder.Build();
+
+app.MapGet("/", async ([FromKeyedServices("entries")]Container container) =>
+{
+    // Add an entry to the database on each request.
+    await container.CreateItemAsync(new Entry(Guid.NewGuid().ToString(), sessionId)).ConfigureAwait(false);
+
+    var entries = new List<Entry>();
+    var iterator = container.GetItemQueryIterator<Entry>(requestOptions: new QueryRequestOptions() { MaxItemCount = 5 });
```

Comment:
@ealsur - will this use query serialization?  If so, it's not going to use the S.T.J serializer until https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4138 ships and your serializer is updated with the new API

Reply:
This call here is effectively a ReadFeed operation (a query without any query text)